src/ThreeDSMethod.res

```diff
@@ -0,0 +1,111 @@
+        let threeDsDataDict =
+          metaDataDict
+          ->Js.Dict.get("threeDSData")
+          ->Belt.Option.flatMap(Js.Json.decodeObject)
+          ->Belt.Option.getWithDefault(Js.Dict.empty())
+        let threeDsUrl =
+          threeDsDataDict
+          ->Js.Dict.get("three_ds_method_details")
+          ->Belt.Option.flatMap(Js.Json.decodeObject)
+          ->Belt.Option.flatMap(x => x->Js.Dict.get("three_ds_method_url"))
+          ->Belt.Option.flatMap(Js.Json.decodeString)
+          ->Belt.Option.getWithDefault("")
+        let threeDsMethodData =
+          threeDsDataDict
+          ->Js.Dict.get("three_ds_method_details")
+          ->Belt.Option.flatMap(Js.Json.decodeObject)
+          ->Belt.Option.flatMap(x => x->Js.Dict.get("three_ds_method_data"))
+          ->Belt.Option.getWithDefault(Js.Dict.empty()->Js.Json.object_)
+        let iframeId = metaDataDict->getString("iframeId", "")
+
+        open Promise
+        PaymentHelpers.threeDsMethod(threeDsUrl, threeDsMethodData, ~optLogger=Some(logger))
+        ->then(res => {
+          mountToInnerHTML(res)
+          resolve(res)
+        })
+        ->then(res => {
+          metadata->Utils.getDictFromJson->Js.Dict.set("3dsMethodComp", "Y"->Js.Json.string)
+          handlePostMessage([
+            ("fullscreen", true->Js.Json.boolean),
+            ("param", `3dsAuth`->Js.Json.string),
+            ("iframeId", iframeId->Js.Json.string),
+            ("metadata", metadata),
+          ])
+          resolve(res)
+        })
+        ->catch(e => {
+          metadata->Utils.getDictFromJson->Js.Dict.set("3dsMethodComp", "N"->Js.Json.string)
+          handlePostMessage([
+            ("fullscreen", true->Js.Json.boolean),
+            ("param", `3dsAuth`->Js.Json.string),
+            ("iframeId", iframeId->Js.Json.string),
+            ("metadata", metadata),
+          ])
+          Js.log("3DS validation failed")
+          reject(e)
+        })
+        ->ignore
+
+        let headersDict =
+          metaDataDict
+          ->getJsonObjectFromDict("headers")
+          ->Js.Json.decodeObject
+          ->Belt.Option.getWithDefault(Js.Dict.empty())
+        let headers = Js.Dict.empty()
+
+        headersDict
+        ->Js.Dict.entries
+        ->Js.Array2.forEach(entries => {
+          let (x, val) = entries
+          Js.Dict.set(headers, x, val->Js.Json.decodeString->Belt.Option.getWithDefault(""))
+        })
+        let _timeExpiry = metaDataDict->getString("expiryTime", "")
+      }
+    }
+    Window.addEventListener("message", handle)
+    Some(() => {Window.removeEventListener("message", handle)})
+  })
+  React.useEffect1(() => {
+    if expiryTime < 1000.0 {
+      Modal.close(setOpenModal)
+    }
+    let intervalID = Js.Global.setInterval(() => {
+      setExpiryTime(prev => prev -. 1000.0)
+    }, 1000)
+    Some(
+      () => {
+        Js.Global.clearInterval(intervalID)
+      },
```

Comment:
maybe remove the wrapping `{}` so the whole return statement can come in a single line.

Reply:
also, now that we have rescript core, you can call setInterval and clearInterval directly. No need to prefix it with `Js.Global.`